default/python/universe_generation/empires.py

```diff
@@ -154,11 +169,15 @@ def find_home_systems_for_min_jump_distance(self, systems_pool, min_jumps):
                 # remove all neighbors from the local pool
                 local_pool -= set(get_systems_within_jumps(member, min_jumps))
 
-            # Count complete misses when number of candidates is not close to the target.
-            if len(candidate) < (self.num_home_systems - 4):
-                num_complete_misses_remaining -= 1
-
             if len(candidate) < self.num_home_systems:
+
+                # Break if the normal distribution of failed candidates is too much smaller than
+                # the target number of systems
+                lens_of_failed_candidates.append(len(candidate))
+                mean_len_of_failed_candidates = sum(lens_of_failed_candidates) / float(len(lens_of_failed_candidates))
+                if mean_len_of_failed_candidates < lowest_continuable_mean_len_of_candidates:
+                    break
```

Comment:
Your ```expected_len_candidate_std``` is based on 3 measurements. You have 2 variables and don't even use one in the final formula. Unless you have some sound theoretical model to accompany that formula, I don't buy it the slightest. It seems arbitrary to me (you could fit up to N-> infinity different functions that would approximate your 3 measurements well enough...). Also I can't quite believe that a Cluster with Low starlanes would yield the same std as an elliptical galaxy with high starlane density.

You write that ```lens_of_failed_candidates``` is roughly a normal distributed variable.
In that case, I think it would be more reasonable to estimate the underlying distribution here by calculating mean and standard deviations and break if (mean + m*std) < num_home_systems with a minimum number of tries...


Even assuming your estimated std was correct, your break condition is completely unstable against statistics. You need some minimum sample size before you break. A single bad first selection can screw you up badly otherwise. So what you probably would want to do is calculate and consider the uncertainty of the mean here as well (i.e. std / sqrt(n)).  But again, I don't think the approach is valid at all... Correctness should definitely go before speed for Universe Generation, imo.